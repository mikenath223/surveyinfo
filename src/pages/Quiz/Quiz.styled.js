import styled from 'styled-components';
import { breakpoints } from 'styles/global';

export const Wrapper = styled.main`
  min-height: 100vh;
  background-color: ${({ theme }) => theme.colors.bottomDropBg};
`

export const Hero = styled.section`
  background: url(/assets/survey.png) center no-repeat;
  background-color: ${({ theme }) => theme.colors.headBg};
  background-position: 90%;
  background-size: 300px;
  padding: 30px 0 0 100px;
  min-height: 300px;

  @media ${breakpoints.sm} {
    background-size: 150px;
    padding: 30px 0 0 30px;
  }
`;

export const HeroText = styled.h1`
  font-size: ${({ theme }) => theme.textSizes.mainText};
  color: ${({ theme }) => theme.colors.mainText};
  font-family: Montserrat;
  font-weight: 700;
  padding-top: 4.5rem;
  text-transform: uppercase;
  margin: 0;

  @media ${breakpoints.md} {
    font-size: 20px;
    padding-top: 7.5rem;
  }
`;

export const HeroSubText = styled.h4`
  font-size: ${({ theme }) => theme.textSizes.subText};
  color: ${({ theme }) => theme.colors.subText};
  font-style: italic;

  @media ${breakpoints.sm} {
    font-size: 14px;
  }
`;

export const DashBoard = styled.section`
  margin: -90px auto 100px auto;
  min-height: 500px;
  max-width: 1200px;
  padding: 20px;
`;

export const InnerSection = styled.div`
  background-color: ${({ theme }) => theme.colors.dashboardBg};
  display: flex;
  max-width: 1200px;
  min-height: 500px;
  border-radius: 20px;
`;

export const BoardAside = styled.div`
  background: url(/assets/mailbox.png) center no-repeat;
  background-color: ${({ theme }) => theme.colors.asideBg};
  background-size: 190px;
  width: 500px;
  min-height: 500px;
  border-radius: 20px 0;

  @media ${breakpoints.md} {
    display: none;
  }
`;

export const FormSection = styled.div`
  display: flex;
  justify-content: center;
  width: 100%
`;