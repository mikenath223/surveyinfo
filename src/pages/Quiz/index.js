import React from 'react';
import { connect } from 'react-redux';
import { useHistory } from 'react-router-dom';
import { withTheme } from 'styled-components';
import {
  Hero,
  HeroText,
  HeroSubText,
  DashBoard,
  InnerSection,
  BoardAside,
  FormSection,
} from './Quiz.styled';
import { setThemeMode, saveQuizResult } from 'store/actions';
import Form from 'components/Form';

const Quiz = ({ theme, setThemeMode, saveQuizResult }) => {
  const history = useHistory();

  const handleSubmit = values => {
    const { message, date, layout, color, text } = values;
    saveQuizResult({ message, date });
    const theme = `${color}_${layout}_${text}_TEXT`;
    setThemeMode(theme);
    history.push('/result');
  }

  return <main>
    <Hero>
      <HeroText>Survey Grill</HeroText>
      <HeroSubText>We invite you to come take our survey</HeroSubText>
    </Hero>
    <DashBoard>
      <InnerSection>
        <BoardAside />
        <FormSection>
          <Form handleSubmit={handleSubmit} />
        </FormSection>
      </InnerSection>
    </DashBoard>
  </main>
}

const mapDispatchToProps = {
  saveQuizResult,
  setThemeMode
}

export default withTheme(connect(null, mapDispatchToProps)(Quiz))