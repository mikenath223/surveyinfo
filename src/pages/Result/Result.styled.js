import styled from 'styled-components';
import { breakpoints } from 'styles/global';
import { Row } from 'react-bootstrap';

export const ResultHeadText = styled.h2`
  font-size: ${({ theme }) => theme.textSizes.mainText};
  color: ${({ theme }) => theme.colors.subText};
  margin: 20px 0;
  border-radius: 20px;
  text-align: center;

  @media ${breakpoints.md} {
    font-size: 20px;
  }
`

export const MainBoard = styled.div`
  background-color: ${({ theme }) => theme.colors.dashboardBg};
  width: 100%;
  border-radius: 20px;
`

export const ResultSection = styled.div`
  display: ${({ theme }) => theme.display};
  justify-content: space-around;
  flex-direction: column;
  grid-template-columns: 1fr 1fr;
  gap: 30px;
  margin: 10px 30px;
`

export const Text = styled.h3`
  text-align: center;
  padding: 5px 0;
  color: ${({ theme }) => theme.colors.introText};
  background-color: ${({ theme }) => theme.colors.resultHead};
  border-radius: 10px;

  &.result {
    white-space: wrap;
    text-align: unset;
    padding-left: 10px;
    background-color: unset;
  }
`

export const Message = styled.p`
  font-size: ${({ theme }) => theme.textSizes.mainText};
  color: ${({ theme }) => theme.colors.mainText};
  background-color: ${({ theme }) => theme.colors.resultBg};
  border-radius: 10px;
  min-height: 300px;
`;

export const Date = styled.p`
  font-size: ${({ theme }) => theme.textSizes.mainText};
  color: ${({ theme }) => theme.colors.mainText};
  background-color: ${({ theme }) => theme.colors.resultBg};
  border-radius: 10px;
  max-height: 150px;
`;

export const BottomLinks = styled(Row)``