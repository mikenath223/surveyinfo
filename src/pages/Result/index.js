import React, { useEffect } from 'react'
import { Link, useHistory } from 'react-router-dom';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import {
  Hero, HeroText, HeroSubText, DashBoard,
  InnerSection, BoardAside, Wrapper
} from 'pages/Quiz/Quiz.styled';
import {
  ResultHeadText, MainBoard, Text,
  ResultSection, Message, Date, BottomLinks
} from './Result.styled';
import ButtonLink from 'components/Button';
import { Col } from 'react-bootstrap';
import Footer from 'components/Footer';

const Result = ({ quizResult: { message, date } }) => {
  const history = useHistory();

  useEffect(() => {
    if (!message || !date) history.push('/')
  }, [message, date])

  return <Wrapper>
    <Hero>
      <HeroText>Survey Grill</HeroText>
      <HeroSubText>We invite you to come take our survey</HeroSubText>
    </Hero>
    <DashBoard>
      <InnerSection>
        <BoardAside />
        <MainBoard>
          <ResultHeadText>Your Result</ResultHeadText>
          <ResultSection>
            <Message>
              <Text>Message</Text>
              <Text as="p" className="result">
                {message}
              </Text>
            </Message>
            <Date>
              <Text>Date</Text>
              <Text as="p" className="result">
                {date}
              </Text>
            </Date>
          </ResultSection>
          <BottomLinks className="mx-2 my-5">
            <Col md={6} className="text-center my-3">
              <ButtonLink as={Link} className="bg-primary p-2 rounded text-white link" to="/">
                Retake Survey
              </ButtonLink>
            </Col>
            <Col md={6} className="text-center pt-3">
              <ButtonLink as="a" className="source bg-danger  p-2 rounded font-weight-bold" rel="noopener noreferrer" target="_blank" href="https://github.com/mikenath223/surveyinfo">
                  Source Code
              </ButtonLink>
            </Col>
          </BottomLinks>
        </MainBoard>
      </InnerSection>
    </DashBoard>
    <Footer/>
  </Wrapper>
}

const mapStateToProps = ({ quizResult }) => ({
  quizResult,
})

Result.propTypes = {
  quizResult: PropTypes.shape({
    message: PropTypes.string,
    date: PropTypes.string
  })
}

export default connect(mapStateToProps, null)(Result)