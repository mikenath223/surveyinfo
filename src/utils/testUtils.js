import React from 'react';
import { ThemeProvider } from 'styled-components';
import { render } from '@testing-library/react';
import { theme } from 'styles/global';

export const withThemeProvider = (component) =>
  render(<ThemeProvider theme={theme}>{component}</ThemeProvider>);

export const withResultThemeProvider = (component) =>
  render(<ThemeProvider theme={theme}>
    <ThemeProvider theme={theme}>{component}</ThemeProvider>
  </ThemeProvider>)