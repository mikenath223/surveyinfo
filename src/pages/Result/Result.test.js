import React from 'react';
import 'jest-styled-components';
import { withProviders, withResultThemeProvider } from 'utils/testUtils';
import Result from './index';
import { Message } from './Result.styled';
import { Wrapper } from 'pages/Quiz/Quiz.styled';

describe('Result page', function () {
  let defaultTheme = 'LIGHT_GRID_LARGE_TEXT';
  let quizResult = { message: 'yes', date: '12/12/20' }

  it('should match snapshot', () => {
    const { container } = withProviders(<Result quizResult={quizResult} />, defaultTheme);

    expect(container).toMatchSnapshot();
  });

  it('should show correct themeMode on result', () => {
    const tree = withResultThemeProvider(<Message />, 'DARK_GRID_LARGE_TEXT').toJSON();

    expect(tree).toHaveStyleRule('background-color', '#121212');
    expect(tree).toHaveStyleRule('color', '#fff')
  })

  it('should show correct themeMode on body', () => {
    const tree = withResultThemeProvider(<Wrapper />, 'DARK_GRID_LARGE_TEXT').toJSON();

    expect(tree).toHaveStyleRule('background-color', '#080808');
  })
});
