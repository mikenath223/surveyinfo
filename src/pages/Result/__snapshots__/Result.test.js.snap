// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Result page should match snapshot 1`] = `
.c0 {
  min-height: 100vh;
  background-color: #EBEBED;
}

.c1 {
  background: url(/assets/survey.png) center no-repeat;
  background-color: #826eef;
  background-position: 90%;
  background-size: 300px;
  padding: 30px 0 0 100px;
  min-height: 300px;
}

.c2 {
  font-size: 40px;
  color: #fff;
  font-family: Montserrat;
  font-weight: 700;
  padding-top: 4.5rem;
  text-transform: uppercase;
  margin: 0;
}

.c3 {
  font-size: 22px;
  color: #000;
  font-style: italic;
}

.c4 {
  margin: -90px auto 100px auto;
  min-height: 500px;
  max-width: 1200px;
  padding: 20px;
}

.c5 {
  background-color: #fff;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  max-width: 1200px;
  min-height: 500px;
  border-radius: 20px;
}

.c6 {
  background: url(/assets/mailbox.png) center no-repeat;
  background-color: silver;
  background-size: 190px;
  width: 500px;
  min-height: 500px;
  border-radius: 20px 0;
}

.c8 {
  font-size: 25px;
  color: #000;
  margin: 50px 0;
  border-radius: 20px;
  text-align: center;
}

.c7 {
  background-color: #fff;
  width: 100%;
  border-radius: 20px;
}

.c9 {
  display: grid;
  -webkit-box-pack: space-around;
  -webkit-justify-content: space-around;
  -ms-flex-pack: space-around;
  justify-content: space-around;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  grid-template-columns: 48% 48%;
  gap: 30px;
  margin: 10px 30px;
}

.c11 {
  font-size: 25px;
  color: #000;
  background-color: #E6E6E6;
  text-align: center;
  border-radius: 10px;
  padding: 5px 0;
  overflow: hidden;
}

.c11.result {
  text-align: unset;
  padding: 10px;
  background-color: unset;
}

.c10 {
  font-size: 25px;
  color: #fff;
  background-color: #C5C5C5;
  border-radius: 10px;
  min-height: 300px;
}

.c12 {
  font-size: 25px;
  color: #fff;
  background-color: #C5C5C5;
  border-radius: 10px;
  max-height: 150px;
}

.c13.source {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #fff;
}

.c13.link {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c14 {
  width: 80%;
  margin: auto;
}

.c16 {
  text-align: center;
  margin: 1.5rem 0;
  font-size: 15px;
  color: #000;
}

.c15 {
  border-top: 1px solid grey;
  margin-top: 4rem;
  margin-bottom: 0;
  opacity: 0.3;
}

@media (max-width:576px) {
  .c1 {
    background-size: 150px;
    padding: 30px 0 0 30px;
  }
}

@media (max-width:768px) {
  .c2 {
    font-size: 20px;
    padding-top: 7.5rem;
  }
}

@media (max-width:576px) {
  .c3 {
    font-size: 14px;
  }
}

@media (max-width:768px) {
  .c6 {
    display: none;
  }
}

@media (max-width:768px) {
  .c8 {
    font-size: 14px;
  }
}

@media (max-width:576px) {
  .c9 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
  }
}

@media (max-width:576px) {
  .c11 {
    font-size: 14px;
  }
}

@media (max-width:768px) {
  .c10 {
    font-size: 14px;
  }
}

@media (max-width:768px) {
  .c12 {
    font-size: 14px;
  }
}

<div>
  <main
    class="c0"
  >
    <section
      class="c1"
    >
      <h1
        class="c2"
      >
        Survey Grill
      </h1>
      <h4
        class="c3"
      >
        We invite you to come take our survey
      </h4>
    </section>
    <section
      class="c4"
    >
      <div
        class="c5"
      >
        <div
          class="c6"
        />
        <div
          class="c7"
        >
          <h2
            class="c8"
          >
            Your Result
          </h2>
          <div
            class="c9"
          >
            <div
              class="c10"
            >
              <h3
                class="c11"
              >
                Message
              </h3>
              <p
                class="c11 result"
              />
            </div>
            <div
              class="c12"
            >
              <h3
                class="c11"
              >
                Date
              </h3>
              <p
                class="c11 result"
              />
            </div>
          </div>
          <div
            class="mx-2 my-5 row"
          >
            <div
              class="text-center my-3 col-md-6"
            >
              <a
                class="c13 bg-primary p-2 rounded text-white link"
                href="/"
              >
                Retake Survey
              </a>
            </div>
            <div
              class="text-center pt-3 col-md-6"
            >
              <a
                class="c13 source bg-danger  p-2 rounded font-weight-bold"
                href="https://github.com/mikenath223/surveyinfo"
                rel="noopener noreferrer"
                target="_blank"
              >
                Source Code
              </a>
            </div>
          </div>
        </div>
      </div>
    </section>
    <div
      class="c14 row"
    >
      <div
        class="col-sm-12"
      >
        <hr
          class="c15"
        />
      </div>
      <div
        class="col-sm-12"
      >
        <div
          class="c16"
        >
          Â© 2020 CheabanTest. All rights reserved.
        </div>
      </div>
    </div>
  </main>
</div>
`;
